### PR TITLE
Initial Auth documentation for Android

### PR DESCRIPTION
**Pull request by undefobj**
Initial Auth documentation with new AWSMobileClient matching iOS documentation layout. Facebook and Google sign-in need to be cleaned up after this PR

### CODE REVIEW

File: android/authentication.md [hunk 1]
```diff
@@ -1,615 +1,704 @@
...
+Creates a new user in your User Pool:
 
-	```
-	❯ User Sign-Up, Sign-In, connected with AWS IAM controls ...
-	```
+```java
+final String username = getInput(R.id.signUpUsername);
+final String password = getInput(R.id.signUpPassword);
+final Map<String, String> attributes = new HashMap<>();
+attributes.put("email", "bimin@amazon.com");
```
Comment (muellerfr) [inline android/authentication.md]: Should we use a generic email address here?

File: android/authentication.md [hunk 2]
```diff
@@ -1,615 +1,704 @@
...
 
-* To cancel the sign-in flow, set `.canCancel(true)`.
+`AWSMobileClient` offers on-demand querying for the "login state" of a user in the application. For instance, you can check if the user is signed-in or not and present an appropriate screen. This is done through a couple of mechanisms:
 
-* To change the font in the sign-in views, use the `fontFamily` method and pass in the string that represents a font family.
+- `isLoggedIn` property defined as a BOOLEAN for the most simple use cases
+- `currentUserState` used for more advanced scenarios, such as determining if the user has Guest credentials, Authenticated with User Pools, has Federated credentials, or has signed out.
 
-* To draw the `backgroundColor` full screen, use `fullScreenBackgroundColor`.
+This allows you to write workflows in your application based on the state of the user and what you would like to present on different screens. The `AWSMobileClient` also offers realtime notifications on user state changes which you can register for in your application using `.addSignInStateListener` as in the code below.
```
Comment (muellerfr) [inline android/authentication.md]: .addSignInStateListener => .addSignInStateListener()

Comment (muellerfr) [reply android/authentication.md]: I think there are a couple more of those in the doc.

Comment (undefobj) [reply android/authentication.md]: Done

File: android/authentication.md (hunk 1 above)
Comment (undefobj) [reply android/authentication.md]: Yeah that's a copy/paste miss, fixed

File: android/authentication.md [hunk 3]
```diff
@@ -1,615 +1,704 @@
...
-                    .build()
-            val signInUI = AWSMobileClient.getInstance().getClient(this@MainActivity, SignInUI::class.java) as SignInUI
-            signInUI.login(this@MainActivity, NextActivity::class.java).authUIConfiguration(config).execute()
-        }.execute()
-    }
-}
 ```
-</div>
-
+minSdkVersion 23
```
Comment (minbi) [inline android/authentication.md]: Prefer for this to be 15. Quick visual inspection may deter people from using mobile client. The IDE or compile will tell them they need 23 when adding the drop-in UI piece.

File: android/authentication.md [hunk 4]
```diff
@@ -1,615 +1,704 @@
...
 
-6. Choose Facebook and then provide your Facebook app ID. To retrieve or create your Facebook app ID, see [Setting Up Facebook Authentication](http://docs.aws.amazon.com/aws-mobile/latest/developerguide/auth-facebook-setup.html).
+When complete run `amplify push` and your `awsconfiguration.json` will work automatically with your updated Cognito settings. The `AWSMobileClient` user session will automatically have permissions configured for Guest/Unauthenticated users upon initialization. 
 
-7. When configuration for Facebook sign-in is complete, the CLI displays a message confirming that you have configured local CLI metadata for this category. You can confirm this by viewing status.
+If you login in your app either using the [Drop-In Auth](#dropinui) or the [direct Auth APIs](#iosapis) then the `AWSMobileClient` user session will transition to an authenticated role.
```
Comment (minbi) [inline android/authentication.md]: iosapis -> androidapis?

File: android/authentication.md [hunk 5]
```diff
@@ -1,615 +1,704 @@
...
-  }.execute()
+![Image](./images/new-facebook-add-platform.png)
 
-}
-```
+8. Choose your platform and provide information about your app that Facebook will use for
+   integration during credential validation.
 
-Choose the run icon (|play|) in Android Studio to build your app and run it on your device/emulator. You should see the ready made sign-in UI for your app. Check out the next steps to learn how to [customize your UI](add-aws-mobile-user-sign-in-customize).
+   `For iOS:`
```
Comment (minbi) [inline android/authentication.md]: This is still an iOS section

File: android/authentication.md [hunk 6]
```diff
@@ -1,615 +1,704 @@
...
-                signin.login(
-                      AuthenticatorActivity.this,
-                      NextActivity.class).execute();
-            }
-        }).execute();
-    }
-}
+* [Create an OAuth Android Client ID](./auth-google-setup#create-an-oauth-android-client-id) (required for all Android apps)
+
+* [Create an OAuth iOS Client ID](./auth-google-setup#create-an-oauth-ios-client-id) (required for all iOS apps)
```
Comment (minbi) [inline android/authentication.md]: is this needed?

File: android/authentication.md [hunk 7]
```diff
@@ -806,192 +793,24 @@ import com.amazonaws.mobile.client.AWSMobileClient;
...
+public class AuthenticatorActivity extends Activity {
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_authenticator);
 
-            override fun onUserSignedOut() {
-                Log.d(TAG, "User signed out")
+        // Add a call to initialize AWSMobileClient
+        AWSMobileClient.getInstance().initialize(this, new AWSStartupHandler() {
```
Comment (minbi) [inline android/authentication.md]: this should be `new Callback<UserStateDetails>()....`

File: android/authentication.md (hunk 3 above)
Comment (undefobj) [reply android/authentication.md]: Agree. Fixed.

File: android/authentication.md (hunk 4 above)
Comment (undefobj) [reply android/authentication.md]: This was conversion from the Margherita repo. Fixed

File: android/authentication.md (hunk 5 above)
Comment (undefobj) [reply android/authentication.md]: Removed but as noted in the PR I need to scrub the Facebook and Google sections as they're not reviewed.

File: android/authentication.md (hunk 6 above)
Comment (undefobj) [reply android/authentication.md]: Per-previous comment, I'm not sure. Let's talk.

**Pull request merged**